synapse/handlers/__init__.py

```diff
@@ -54,7 +55,12 @@ def __init__(self, hs):
         self.directory_handler = DirectoryHandler(hs)
         self.typing_notification_handler = TypingNotificationHandler(hs)
         self.admin_handler = AdminHandler(hs)
+        asapi = ApplicationServiceApi(hs)
         self.appservice_handler = ApplicationServicesHandler(
-            hs, ApplicationServiceApi(hs)
+            hs, asapi, AppServiceScheduler(
+                clock=hs.get_clock(),
+                store=hs.get_datastore(),
+                as_api=asapi
+            )
```

Comment:
It's more that everywhere else uses the `HomeServer` object to do DI, so unless there is a compelling reason not to I would prefer we were consistent.

In fact, we should be adding all three `ApplicationServiceApi`, `AppServiceScheduler` and `ApplicationServicesHandler` to `HomeServer`. Half the point of the `HomeServer` object is to have all the DI in one place.